crates/stages/src/stages/hashing_storage.rs

```diff
@@ -160,33 +160,40 @@ impl<DB: Database> Stage<DB> for StorageHashingStage {
                 // Assumption we are okay with is that plain state represent
                 // `previous_stage_progress` state.
                 .map(|(address, storage)| {
-                    storage
-                        .into_iter()
-                        .map(|key| {
-                            plain_storage
-                                .seek_by_key_subkey(address, key)
-                                .map(|ret| (keccak256(key), ret.map(|e| e.value)))
-                        })
-                        .collect::<Result<BTreeMap<_, _>, _>>()
-                        .map(|storage| (keccak256(address), storage))
+                    let res = (
+                        keccak256(address),
+                        storage
+                            .into_iter()
+                            .map(|key| {
+                                Ok::<Option<_>, reth_db::Error>(
+                                    plain_storage
+                                        .seek_by_key_subkey(address, key)?
+                                        .filter(|v| v.key == key)
+                                        .map(|ret| (keccak256(key), ret.value)),
+                                )
+                            })
+                            .collect::<Result<Vec<Option<_>>, _>>()?
+                            .into_iter()
+                            .flatten()
+                            .collect::<BTreeMap<_, _>>(),
+                    );
+                    Ok::<_, reth_db::Error>(res)
                 })
                 .collect::<Result<BTreeMap<_, _>, _>>()?
                 .into_iter()
                 // Hash the address and key and apply them to HashedStorage (if Storage is None
                 // just remove it);
-                .try_for_each(|(address, storage)| {
+                .try_for_each(|(hashed_address, storage)| {
                     storage.into_iter().try_for_each(|(key, val)| -> Result<(), StageError> {
                         if hashed_storage
-                            .seek_by_key_subkey(address, key)?
+                            .seek_by_key_subkey(hashed_address, key)?
                             .filter(|entry| entry.key == key)
                             .is_some()
                         {
                             hashed_storage.delete_current()?;
                         }
 
-                        if let Some(value) = val {
-                            hashed_storage.upsert(address, StorageEntry { key, value })?;
-                        }
+                        hashed_storage.upsert(hashed_address, StorageEntry { key, value: val })?;
```

Comment:
Hey @joshieDo is this statement correct: if the storage key is not found (Is zero) it shouldn't be found in hashed table?
I did some refactoring and this came up. `.filter(|v| v.key == key)` was a bug but this shouldn't be ( I added unwrap_or_default that is why it compares to U256::ZERO)
want to double check